Add initial test owners for framework post-submit tests

keyonghan opened the pull request:
## Description
This adds a new TESTOWNERS file to maintain up-to-date ownership info of tests belonging to this repository. 

It follows format:
`<Test builder name> @<team leader> @<sub-team label>`.
For example: `Linux analyzer_benchmark @zanderso @tool`

This file initially will be used as a reference for flaky bugs assignment, but can be combined with [CODEOWNERS](https://github.com/flutter/flutter/blob/master/CODEOWNERS) in the future if we want to enforce review for different test changes.

## Tests
None here, but a pre-submit check will be added to enforce test ownership info.

keyonghan committed:
add initial test owners

keyonghan committed:
remove trailing space

keyonghan commented:
A friendly ping.

Hixie commented on TESTOWNERS:
```diff
@@ -0,0 +1,219 @@
+# for further triage.
+
+# Linux Android DeviceLab tests
+Linux analyzer_benchmark @zanderso @tool
```
What's the namespace for `@tool`?

keyonghan replied:
The last entry is the sub-team label which will be used for their bug triage.
Since we do not have any github teams created, does it make sense to you to add that label here? Or shall we create corresponding sub teams?

keyonghan replied:
Used `@flutter/<subteam>` instead to follow [CODEOWNERS format](https://docs.github.com/en/github/creating-cloning-and-archiving-repositories/about-code-owners).

keyonghan committed:
change to use paths

keyonghan committed:
update sub team

keyonghan commented:
> Can we make these be paths to the actual tests? Right now it's not clear what these names mean. (I assume it's the shard names?) Many of the tests are also listed several times, e.g. if a test runs on Windows, Linux, and macOS, it's listed three times even though it's just one test. Using the shard names also means that what should be arbitrary infrastructure work (e.g. splitting a set of 1000 tests that come from three directories into 4 shards instead of 5 suddenly changes who owns specific tests.)

As chatted offline, I updated to use paths of test entry files for all DeviceLab and ad hoc ones.

As for framework host only tests with shards (called via [`test.dart`](https://github.com/flutter/flutter/blob/master/dev/bots/test.dart)), I do not have a clear answer for their test files/paths for now. Created a tracking issue https://github.com/flutter/flutter/issues/82068 to follow up later and unblock the reduce flakiness workflow.

keyonghan committed:
add ownership for legacy add2app tests

keyonghan commented:
> I would have expected entries for all the tests in `packages/flutter/test`, `packages/flutter_tools/test` and so on also.

Those are covered in the TODO: https://github.com/flutter/flutter/issues/82068
Additionally, added cirrus and pre-submit LUCI ones in the tracking issue.

> Please make sure to get LGTMs from anyone you're listing explicitly. (Failing that, it's probably fine to just list the teams and cc the TLs manually when using this list, rather than have them listed here.)

Added TLs for review.

ferhatb reviewed (approved):
LGTM for web

stuartmorgan commented on TESTOWNERS:
```diff
@@ -0,0 +1,201 @@
+
+# Host only DeviceLab tests
+/dev/devicelab/bin/tasks/build_aar_module_test.dart @zanderso @flutter/tool
+/dev/devicelab/bin/tasks/gradle_desugar_classes_test.dart @stuartmorgan @flutter/plugin
```
Are you sure this is plugins?

keyonghan replied:
Good catch! I revisited my update history, and this one should have been falling in those that need double check. Based on communication with @zanderso , those remaining devicelab host only tests will be assigned to `tools`. Have updated.

stuartmorgan-g reviewed (approved):
LGTM for plugin with one question

keyonghan committed:
update gradle_desugar_classes_test

HansMuller reviewed (approved):
LGTM for framework

keyonghan commented:
Thanks all for the review and LGTMs!
